mgmt/src/models/internal/natconfig/collapse.rs

```diff
@@ -61,7 +61,7 @@ fn collapse_prefix_lists(
             // exclusion prefixes.
             if exclude.covers(prefix) {
                 result.remove(prefix);
-                break;
+                continue;
```

Comment:
Good catch, thanks! How did you catch it?

```
Prefixes:
10.0.0.0/24
10.0.1.0/24

Excludes:
10.0.0.0/16
```

Note that, unless I'm mistaken, this case cannot produce a valid `VpcExpose`. This is because [we check](https://github.com/githedgehog/dataplane/blob/main/mgmt/src/models/external/overlay/vpcpeering.rs#L112) that all exclusion prefixes are contained by at least one allowed prefixes, and [we also check](https://github.com/githedgehog/dataplane/blob/main/mgmt/src/models/external/overlay/vpcpeering.rs#L100) that allowed prefixes don't overlap. In your example, to have a valid `VpcExpose`, `10.0.0.0/16` should be contained by some allowed prefix, which would then overlap with both `10.0.0.0/24` and `10.0.1.0/24`.

Still a good idea to fix the algo :+1:

Reply:
Fuzz tests were failing and I thought it was the new code, turns out it was an existing bug.